lib/rules/jsx-key.js

```diff
@@ -61,9 +68,34 @@ module.exports = {
       return body.filter((item) => item.type === 'ReturnStatement')[0];
     }
 
+    function isKeyAfterSpread(attributes) {
+      let keyAttributeIndex = -1;
+      let spreadAttributeIndex = -1;
+
+      attributes.forEach((attribute, i) => {
+        if (attribute.type === 'JSXSpreadAttribute') {
+          spreadAttributeIndex = i;
+        }
+
+        if (attribute.type === 'JSXAttribute') {
+          if (propName(attribute) === 'key') {
+            keyAttributeIndex = i;
+          }
+        }
+      });
+
+      return spreadAttributeIndex !== -1 && keyAttributeIndex > spreadAttributeIndex;
```

Comment:
```suggestion
      var hasFoundSpread = false;
      return attributes.some((attribute) => {
        if (attribute.type === 'JSXSpreadAttribute') {
          hasFoundSpread = true;
          return false;
        }
        if (attribute.type !== 'JSXAttribute') {
          return false;
        }
        return !hasFoundSpread || propName(attribute) === 'key';
      });
```
what about this alternative?

Reply:
`hasFoundSpread && propName(attribute) === 'key'` may be clear

Reply:
Sure, in that case you'd need to negate the overarching `.some`.